examples/sn-dms-demo/cypress/integration/documents/documents.spec.ts

```diff
@@ -0,0 +1,166 @@
+import Chance = require('chance')
+import { resources } from '../../../src/assets/resources'
+import {
+  contextMenuItems,
+  createNewFileName,
+  moveToFolderAndCheckIfFileExists,
+  newMenuItems,
+  openContextMenuItem,
+  openNew,
+  registerNewUser,
+  selectPathInListPicker,
+  uploadNewFileAndOpenContextMenuItem,
+} from '../../support/documents'
+
+context('The documents page', () => {
+  let currentUser = { email: '', password: '' }
+
+  const loginOrRegister = (user: { email: string; password: string }) => {
+    cy.login(user.email, user.password).then(isSuccesful => {
+      if (!isSuccesful) {
+        currentUser = registerNewUser()
+        cy.login(currentUser.email, currentUser.password)
+      }
+    })
+  }
+
+  before(async () => {
+    if (process.env.CI) {
+      currentUser = registerNewUser()
+    }
+  })
+
+  beforeEach(() => {
+    // Save current user locally to reduce user creation
+    if (!process.env.CI) {
+      cy.task('getCurrentUser', '../fixtures/currentUser.json')
+        .then(user => (currentUser = user))
+        .then(() => loginOrRegister(currentUser))
+    } else {
+      loginOrRegister(currentUser)
+    }
+  })
+
+  it('header contains the logged in users avatar', () => {
+    cy.visit('#/documents')
+
+    cy.url().should('include', '/documents')
+
+    cy.get(`div[aria-label="${currentUser.email}"]`).should('exist')
+  })
+
+  describe('left side menu', () => {
+    it('should contain new and upload buttons', () => {
+      cy.contains('span[role=button]', resources.UPLOAD_BUTTON_TITLE)
+      cy.contains('span[role=button]', resources.ADD_NEW)
+    })
+
+    it('should contain new document, sheet, slide, text, folder buttons', () => {
+      cy.contains('[data-cy=appbar]', 'Document library').should('exist')
+      cy.contains(resources.ADD_NEW).click()
+      newMenuItems.forEach(item => cy.contains('li[role=menuitem]', `${resources.ADD_NEW} ${item.name}`))
+    })
+
+    it(`creating a new item should show succes notification and can be found in the grid`, () => {
+      cy.contains('Document library').should('exist')
+      newMenuItems.forEach(item => {
+        openNew(item.name)
+        const displayName = Chance().word()
+        cy.get('#DisplayName').type(displayName + '{enter}')
+        cy.contains(displayName + item.ext + ' ' + resources.CREATE_CONTENT_SUCCESS_MESSAGE).should('exist')
+        cy.contains(displayName + item.ext).should('exist')
+        cy.get('[aria-label="Close"]').click()
+      })
+    })
+  })
+
+  it('rename document should work', () => {
+    const fileName = createNewFileName()
+    const newName = createNewFileName()
+    uploadNewFileAndOpenContextMenuItem(currentUser.email, fileName, contextMenuItems.rename)
+    // wait for input to be focused
+    cy.wait(1000)
```

Comment:
In the future, can we refactor these discrete wait values with something [similar](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#2-A-set-timeout-is-reached) method in a separate issue? (more on this in the [best practices](https://docs.cypress.io/guides/references/best-practices.html#Unnecessary-Waiting))

Reply:
We should check this later. I remember that I've added this wait because it find the element too fast and the logic in the rename component does a rerender after 200 or 300 ms.